src/testing/vortex/faulty_network.zig

```diff
@@ -462,6 +462,10 @@ const Proxy = struct {
     accept_fd: std.posix.socket_t,
     mapping: Mapping,
     connections: [constants.connections_count_max]Connection,
+
+    fn deinit(proxy: *Proxy) void {
+        proxy.io.close_socket(proxy.accept_fd);
```

Comment:
Could you also add `proxy.* = undefined` to help making improper usage after deinit fail loudly?